CMakeLists.txt

```diff
@@ -59,12 +55,23 @@ if(MSVC)
     set(CMAKE_DEBUG_POSTFIX "d")
 endif()
 
+# Build position independent code
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```

Comment:
Since this project is also a tutorial, I'd like to give room to comments to explain why we're setting this feature. For example, we could add something like:

> Position Independent Code (PIC) is commonly used for shared libraries so that the same shared library code can be loaded in each program address space in a location where it will not overlap with any other uses of such memory. In particular, this option avoids problems occurring when a process wants to load more than one shared library at the same virtual address. Since shared libraries cannot predict where other shared libraries could be loaded, this is an unavoidable problem with the traditional shared library concept. Generating position-independent code is often the default behavior for most modern compilers.
Further details on PIC can be found here: https://eli.thegreenplace.net/2011/11/03/position-independent-code-pic-in-shared-libraries/


which is elaborated from Wikipedia. If you have precise hints about this command please feel free to add or modify the description.